apps/api/test/functional/templates.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { faker } from "@faker-js/faker";
+import * as fs from "node:fs";
+import * as path from "node:path";
+
+import { app } from "@src/app";
+import type { Category } from "@src/template/types/template";
+import { env } from "@src/utils/env";
+
+const fakeHeaders = {
+  headers: {
+    "x-ratelimit-remaining": 100
+  }
+};
+
+env.GITHUB_PAT = "ghp_1234567890";
+
+jest.mock("@octokit/rest", () => {
```

Comment:
suggestion(non-blocking): avoid `jest.mock` usage in DI based systems. If you still need to use`jest.mock` in DI system, it means that some service doesn't properly encapsulate its dependencies

Reply:
faking external resource is the right way to test the service. The actual issue is how we do it. `jest.mock` is a hacky way to do it. You can provide a mock for github service via DI container.

To simplify its definition, you can pass it as an optional dependency with default value which equal to `Octocit` instance. You can take a look at this example with feature flags -> https://github.com/akash-network/console/blob/main/apps/api/src/core/services/feature-flags/feature-flags.service.ts#L19